android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/DVCLifecycleCallbacks.kt

```diff
@@ -0,0 +1,70 @@
+package com.devcycle.sdk.android.eventsource
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import timber.log.Timber
+
+class DVCLifecycleCallbacks(eventSource: EventSource, customHandler: Handler? = null) : Application.ActivityLifecycleCallbacks {
+    private val CHECK_DELAY: Long = 800
+
+    private val eventSource = eventSource
+    private val listenerThread: HandlerThread = HandlerThread("DVCForegroundListener")
+    private var handler: Handler? = customHandler
+    private var paused = false
+    private var foreground = true
+    private var check: Runnable? = null
+
+    init {
+        if (handler == null) {
+            listenerThread.start()
+            handler = Handler(listenerThread.looper)
+        }
+    }
+    override fun onActivityResumed(activity: Activity) {
+        paused = false
+        if (check != null) {
+            handler!!.removeCallbacks(check!!) // stop check from running if the timer is still running
+            check = null
+        }
+        if (!foreground) {
+            Timber.d("Restarting connection")
+            eventSource.start()
```

Comment:
this needs to make a config call as well

Reply:
good point, added a callback to trigger refetchConfig